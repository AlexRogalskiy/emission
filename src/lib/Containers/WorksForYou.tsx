import { Box, Flex, Separator, Theme } from "@artsy/palette"
import { WorksForYou_query } from "__generated__/WorksForYou_query.graphql"
import Spinner from "lib/Components/Spinner"
import { ZeroState } from "lib/Components/States/ZeroState"
import Notification from "lib/Components/WorksForYou/Notification"
import colors from "lib/data/colors"
import { PAGE_SIZE } from "lib/data/constants"
import { get } from "lib/utils/get"
import { isCloseToBottom } from "lib/utils/isCloseToBottom"
import React from "react"
import {
  FlatList,
  NativeModules,
  RefreshControl,
  ScrollView,
  StyleSheet,
  TextStyle,
  View,
  ViewStyle,
} from "react-native"
import { createPaginationContainer, graphql, RelayPaginationProp } from "react-relay"
import Events from "../NativeModules/Events"

interface Props {
  relay: RelayPaginationProp
  query: WorksForYou_query
}

interface State {
  isRefreshing: boolean
  loadingContent: boolean
}

export class WorksForYou extends React.Component<Props, State> {
  currentScrollOffset?: number = 0

  constructor(props: Props) {
    super(props)

    this.state = {
      isRefreshing: false,
      loadingContent: false,
    }
  }

  componentDidMount() {
    // Update read status in gravity
    NativeModules.ARTemporaryAPIModule.markNotificationsRead(error => {
      if (error) {
        console.warn(error)
      } else {
        Events.postEvent({
          name: "Notifications read",
          source_screen: "works for you page",
        })
      }
    })
  }

  fetchNextPage = () => {
    if (!this.props.relay.hasMore() || this.props.relay.isLoading()) {
      return
    }
    this.setState({ loadingContent: true }, () => {
      this.props.relay.loadMore(PAGE_SIZE, error => {
        if (error) {
          // FIXME: Handle error
          console.error("WorksForYou.tsx", error.message)
        }

        this.setState({ loadingContent: false })
      })
    })
  }

  handleRefresh = () => {
    this.setState({ isRefreshing: true })
    this.props.relay.refetchConnection(PAGE_SIZE, error => {
      if (error) {
        // FIXME: Handle error
        console.error("WorksForYou.tsx #handleRefresh", error.message)
      }
      this.setState({ isRefreshing: false })
    })
  }

  render() {
    const notifications = get(this.props, props => props.query.me.followsAndSaves.notifications.edges)
    const hasNotifications = notifications.length

    /* If showing the empty state, the ScrollView should have a {flex: 1} style so it can expand to fit the screen.
       otherwise, it should not use any flex growth.
    */
    return (
      <ScrollView
        contentContainerStyle={hasNotifications ? {} : styles.container}
        onScroll={isCloseToBottom(this.fetchNextPage)}
        scrollEventThrottle={100}
        refreshControl={<RefreshControl refreshing={this.state.isRefreshing} onRefresh={this.handleRefresh} />}
      >
        <View style={{ flex: 1 }}>{hasNotifications ? this.renderNotifications() : this.renderEmptyState()}</View>
      </ScrollView>
    )
  }

  renderNotifications() {
    const { loadingContent } = this.state
    const notifications = get(this.props, props => props.query.me.followsAndSaves.notifications.edges)

    return (
      <Theme>
        <FlatList
          data={notifications}
          keyExtractor={item => item.node.id}
          scrollEnabled={false}
          renderItem={data => {
            return <Notification notification={data.item.node} />
          }}
          ItemSeparatorComponent={() => (
            <Box px={2}>
              <Separator />
            </Box>
          )}
          ListFooterComponent={() => (
            <>
              {loadingContent && (
                <Box p={2} style={{ height: 50 }}>
                  <Flex style={{ flex: 1 }} flexDirection="row" justifyContent="center">
                    <Spinner />
                  </Flex>
                </Box>
              )}
            </>
          )}
        />
      </Theme>
    )
  }

  renderEmptyState() {
    return (
      <ZeroState
        title="You havenâ€™t followed any artists yet."
        subtitle="Follow artists to see new works that have been added to Artsy"
      />
    )
  }
}

interface Styles {
  container: ViewStyle
  title: TextStyle
  separator: ViewStyle
}

const styles = StyleSheet.create<Styles>({
  container: {
    flex: 1,
  },
  title: {
    marginTop: 20,
    fontSize: 20,
  },
  separator: {
    height: 1,
    backgroundColor: colors["gray-regular"],
  },
})

const WorksForYouContainer = createPaginationContainer(
  WorksForYou,
  {
    query: graphql`
      fragment WorksForYou_query on Query
        @argumentDefinitions(
          count: { type: "Int", defaultValue: 10 }
          cursor: { type: "String" }
          sort: { type: "ArtworkSorts" }
        ) {
        me {
          followsAndSaves {
            notifications: bundledArtworksByArtistConnection(sort: PUBLISHED_AT_DESC, first: $count, after: $cursor)
              @connection(key: "WorksForYou_notifications") {
              pageInfo {
                hasNextPage
                endCursor
              }
              edges {
                node {
                  id
                  ...Notification_notification
                }
              }
            }
          }
        }
      }
    `,
  },
  {
    direction: "forward",
    getConnectionFromProps(props) {
      return props.query.me.followsAndSaves.notifications
    },
    getFragmentVariables(prevVars, totalCount) {
      return {
        ...prevVars,
        count: totalCount,
      }
    },
    getVariables(_props, { count, cursor }, fragmentVariables) {
      return {
        // in most cases, for variables other than connection filters like
        // `first`, `after`, etc. you may want to use the previous values.
        ...fragmentVariables,
        count,
        cursor,
      }
    },
    query: graphql`
      query WorksForYouQuery($count: Int!, $cursor: String) {
        ...WorksForYou_query @arguments(count: $count, cursor: $cursor)
      }
    `,
  }
)

export default WorksForYouContainer
