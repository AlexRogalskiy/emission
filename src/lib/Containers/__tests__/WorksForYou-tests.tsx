import React from "react"
import { NativeModules } from "react-native"
import * as renderer from "react-test-renderer"

import { renderWithLayout } from "../../tests/renderWithLayout"
import { WorksForYou } from "../WorksForYou"

import { Theme } from "@artsy/palette"

beforeAll(() => {
  NativeModules.ARTemporaryAPIModule = { markNotificationsRead: jest.fn() }
  WorksForYou.prototype.componentDidUpdate = () => {
    return null
  }
})

describe("with notifications", () => {
  it("updates the notification count", () => {
    const query = notificationsResponse().query
    renderer
      .create(
        <Theme>
          <WorksForYou query={query as any} relay={null} />
        </Theme>
      )
      .toJSON()
    expect(NativeModules.ARTemporaryAPIModule.markNotificationsRead).toBeCalled()
  })

  it("lays out correctly on small screens", () => {
    const query = notificationsResponse().query
    const component = renderWithLayout(<WorksForYou query={query as any} relay={null} />, { width: 100 })
    expect(component).toMatchSnapshot()
  })

  it("lays out correctly on larger screens", () => {
    const query = notificationsResponse().query
    const component = renderWithLayout(<WorksForYou query={query as any} relay={null} />, { width: 700 })
    expect(component).toMatchSnapshot()
  })
})

describe("without notifications", () => {
  it("lays out correctly on small screens", () => {
    const query = emptyStateResponse().query
    const component = renderWithLayout(<WorksForYou query={query as any} relay={null} />, { width: 100 })
    expect(component).toMatchSnapshot()
  })

  it("lays out correctly on larger screens", () => {
    const query = emptyStateResponse().query
    const component = renderWithLayout(<WorksForYou query={query as any} relay={null} />, { width: 700 })
    expect(component).toMatchSnapshot()
  })
})

interface NotificationsResponse {
  query: {
    me: {
      followsAndSaves: {
        notifications: {
          pageInfo: {
            hasNextPage: boolean
            endCursor: string
          }
          edges: Array<{
            node: {
              artists: string
              summary: string
              artworks: { edges: [{ title: string }] }
              image: {
                resized: {
                  url: string
                }
              }
            }
          }>
        }
      }
    }

    selectedArtist?: any
  }
}

const notificationsResponse = () => {
  return {
    query: {
      me: {
        followsAndSaves: {
          notifications: {
            pageInfo: {
              hasNextPage: true,
              endCursor: null,
            },
            edges: [
              {
                node: {
                  artists: "Jean-Michel Basquiat",
                  summary: "1 Work Added",
                  artworks: { edges: [{ title: "Anti-Product Postcard" }] },
                  image: {
                    resized: {
                      url: "cloudfront.url",
                    },
                  },
                },
              },
              {
                node: {
                  artists: "Ana Mendieta",
                  summary: "2 Works Added",
                  artworks: { edges: [{ title: "CorazÃ³n de Roca con Sangre" }, { title: "Butterfly" }] },
                  image: {
                    resized: {
                      url: "cloudfront.url",
                    },
                  },
                },
              },
            ],
          },
        },
      },
    },
  } as NotificationsResponse
}

const emptyStateResponse = () => {
  return {
    query: {
      me: {
        followsAndSaves: {
          notifications: {
            pageInfo: {
              hasNextPage: true,
            },
            edges: [],
          },
        },
      },
    },
  }
}
