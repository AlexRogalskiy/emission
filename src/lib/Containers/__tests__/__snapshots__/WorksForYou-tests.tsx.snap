// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with notifications lays out correctly on larger screens 1`] = `
<RCTScrollView
  contentContainerStyle={Object {}}
  onScroll={[Function]}
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={false}
    />
  }
  scrollEventThrottle={100}
>
  <RCTRefreshControl />
  <View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RCTScrollView
        ItemSeparatorComponent={[Function]}
        ListFooterComponent={[Function]}
        data={
          Array [
            Object {
              "node": Object {
                "artists": "Jean-Michel Basquiat",
                "artworks": Object {
                  "edges": Array [
                    Object {
                      "title": "Anti-Product Postcard",
                    },
                  ],
                },
                "image": Object {
                  "resized": Object {
                    "url": "cloudfront.url",
                  },
                },
                "summary": "1 Work Added",
              },
            },
            Object {
              "node": Object {
                "artists": "Ana Mendieta",
                "artworks": Object {
                  "edges": Array [
                    Object {
                      "title": "CorazÃ³n de Roca con Sangre",
                    },
                    Object {
                      "title": "Butterfly",
                    },
                  ],
                },
                "image": Object {
                  "resized": Object {
                    "url": "cloudfront.url",
                  },
                },
                "summary": "2 Works Added",
              },
            },
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEnabled={false}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "marginLeft": 20,
                  "marginRight": 20,
                  "marginTop": 20,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "cloudfront.url",
                    }
                  }
                  style={
                    Object {
                      "alignSelf": "center",
                      "backgroundColor": "#f8f8f8",
                      "borderRadius": 20,
                      "height": 40,
                      "marginRight": 10,
                      "width": 40,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "alignSelf": "center",
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 12,
                        },
                        Object {
                          "fontFamily": "AvantGardeGothicITC",
                        },
                        Object {
                          "fontSize": 12,
                        },
                      ]
                    }
                  >
                    JEAN-MICHEL BASQUIAT
                  </Text>
                  <Text
                    numberOfLines={1}
                    style={
                      Array [
                        Object {
                          "fontSize": 17,
                        },
                        Object {
                          "color": "#666666",
                          "fontSize": 14,
                          "marginTop": 2,
                        },
                        Object {
                          "fontFamily": "AGaramondPro-Regular",
                        },
                      ]
                    }
                  >
                    1 Work Added
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "marginBottom": 20,
                    "marginTop": 20,
                  }
                }
              >
                <View
                  onLayout={[Function]}
                >
                  <View
                    accessibilityLabel="Artworks Content View"
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  />
                </View>
              </View>
            </View>
            <View
              px={2}
              style={
                Array [
                  Object {
                    "paddingLeft": 20,
                    "paddingRight": 20,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderBottomWidth": 0,
                      "borderColor": "#E5E5E5",
                      "borderStyle": "solid",
                      "borderWidth": 1,
                      "width": "100%",
                    },
                  ]
                }
                width="100%"
              />
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "marginLeft": 20,
                  "marginRight": 20,
                  "marginTop": 20,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "cloudfront.url",
                    }
                  }
                  style={
                    Object {
                      "alignSelf": "center",
                      "backgroundColor": "#f8f8f8",
                      "borderRadius": 20,
                      "height": 40,
                      "marginRight": 10,
                      "width": 40,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "alignSelf": "center",
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 12,
                        },
                        Object {
                          "fontFamily": "AvantGardeGothicITC",
                        },
                        Object {
                          "fontSize": 12,
                        },
                      ]
                    }
                  >
                    ANA MENDIETA
                  </Text>
                  <Text
                    numberOfLines={1}
                    style={
                      Array [
                        Object {
                          "fontSize": 17,
                        },
                        Object {
                          "color": "#666666",
                          "fontSize": 14,
                          "marginTop": 2,
                        },
                        Object {
                          "fontFamily": "AGaramondPro-Regular",
                        },
                      ]
                    }
                  >
                    2 Works Added
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "marginBottom": 20,
                    "marginTop": 20,
                  }
                }
              >
                <View
                  onLayout={[Function]}
                >
                  <View
                    accessibilityLabel="Artworks Content View"
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
          />
        </View>
      </RCTScrollView>
    </View>
  </View>
</RCTScrollView>
`;

exports[`with notifications lays out correctly on small screens 1`] = `
<RCTScrollView
  contentContainerStyle={Object {}}
  onScroll={[Function]}
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={false}
    />
  }
  scrollEventThrottle={100}
>
  <RCTRefreshControl />
  <View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RCTScrollView
        ItemSeparatorComponent={[Function]}
        ListFooterComponent={[Function]}
        data={
          Array [
            Object {
              "node": Object {
                "artists": "Jean-Michel Basquiat",
                "artworks": Object {
                  "edges": Array [
                    Object {
                      "title": "Anti-Product Postcard",
                    },
                  ],
                },
                "image": Object {
                  "resized": Object {
                    "url": "cloudfront.url",
                  },
                },
                "summary": "1 Work Added",
              },
            },
            Object {
              "node": Object {
                "artists": "Ana Mendieta",
                "artworks": Object {
                  "edges": Array [
                    Object {
                      "title": "CorazÃ³n de Roca con Sangre",
                    },
                    Object {
                      "title": "Butterfly",
                    },
                  ],
                },
                "image": Object {
                  "resized": Object {
                    "url": "cloudfront.url",
                  },
                },
                "summary": "2 Works Added",
              },
            },
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEnabled={false}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "marginLeft": 20,
                  "marginRight": 20,
                  "marginTop": 20,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "cloudfront.url",
                    }
                  }
                  style={
                    Object {
                      "alignSelf": "center",
                      "backgroundColor": "#f8f8f8",
                      "borderRadius": 20,
                      "height": 40,
                      "marginRight": 10,
                      "width": 40,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "alignSelf": "center",
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 12,
                        },
                        Object {
                          "fontFamily": "AvantGardeGothicITC",
                        },
                        Object {
                          "fontSize": 12,
                        },
                      ]
                    }
                  >
                    JEAN-MICHEL BASQUIAT
                  </Text>
                  <Text
                    numberOfLines={1}
                    style={
                      Array [
                        Object {
                          "fontSize": 17,
                        },
                        Object {
                          "color": "#666666",
                          "fontSize": 14,
                          "marginTop": 2,
                        },
                        Object {
                          "fontFamily": "AGaramondPro-Regular",
                        },
                      ]
                    }
                  >
                    1 Work Added
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "marginBottom": 20,
                    "marginTop": 20,
                  }
                }
              >
                <View
                  onLayout={[Function]}
                >
                  <View
                    accessibilityLabel="Artworks Content View"
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  />
                </View>
              </View>
            </View>
            <View
              px={2}
              style={
                Array [
                  Object {
                    "paddingLeft": 20,
                    "paddingRight": 20,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "borderBottomWidth": 0,
                      "borderColor": "#E5E5E5",
                      "borderStyle": "solid",
                      "borderWidth": 1,
                      "width": "100%",
                    },
                  ]
                }
                width="100%"
              />
            </View>
          </View>
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              style={
                Object {
                  "marginLeft": 20,
                  "marginRight": 20,
                  "marginTop": 20,
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "cloudfront.url",
                    }
                  }
                  style={
                    Object {
                      "alignSelf": "center",
                      "backgroundColor": "#f8f8f8",
                      "borderRadius": 20,
                      "height": 40,
                      "marginRight": 10,
                      "width": 40,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "alignSelf": "center",
                      "flex": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "fontSize": 12,
                        },
                        Object {
                          "fontFamily": "AvantGardeGothicITC",
                        },
                        Object {
                          "fontSize": 12,
                        },
                      ]
                    }
                  >
                    ANA MENDIETA
                  </Text>
                  <Text
                    numberOfLines={1}
                    style={
                      Array [
                        Object {
                          "fontSize": 17,
                        },
                        Object {
                          "color": "#666666",
                          "fontSize": 14,
                          "marginTop": 2,
                        },
                        Object {
                          "fontFamily": "AGaramondPro-Regular",
                        },
                      ]
                    }
                  >
                    2 Works Added
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "marginBottom": 20,
                    "marginTop": 20,
                  }
                }
              >
                <View
                  onLayout={[Function]}
                >
                  <View
                    accessibilityLabel="Artworks Content View"
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  />
                </View>
              </View>
            </View>
          </View>
          <View
            onLayout={[Function]}
          />
        </View>
      </RCTScrollView>
    </View>
  </View>
</RCTScrollView>
`;

exports[`without notifications lays out correctly on larger screens 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flex": 1,
    }
  }
  onScroll={[Function]}
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={false}
    />
  }
  scrollEventThrottle={100}
>
  <RCTRefreshControl />
  <View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "height": "100%",
              "justifyContent": "center",
              "marginLeft": 20,
              "marginRight": 20,
              "paddingBottom": 30,
              "paddingLeft": 30,
              "paddingRight": 30,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "ReactNativeAGaramondPro-Regular",
                "fontSize": 30,
                "marginLeft": 10,
                "marginRight": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          You havenât followed any artists yet.
        </Text>
        <Text
          numberOfLines={0}
          style={
            Array [
              Object {
                "fontFamily": "ReactNativeAGaramondPro-Regular",
                "fontSize": 18,
                "marginTop": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          Follow artists to see new works that have been added to Artsy
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`without notifications lays out correctly on small screens 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flex": 1,
    }
  }
  onScroll={[Function]}
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={false}
    />
  }
  scrollEventThrottle={100}
>
  <RCTRefreshControl />
  <View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "height": "100%",
              "justifyContent": "center",
              "marginLeft": 20,
              "marginRight": 20,
              "paddingBottom": 30,
              "paddingLeft": 30,
              "paddingRight": 30,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "ReactNativeAGaramondPro-Regular",
                "fontSize": 30,
                "marginLeft": 10,
                "marginRight": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          You havenât followed any artists yet.
        </Text>
        <Text
          numberOfLines={0}
          style={
            Array [
              Object {
                "fontFamily": "ReactNativeAGaramondPro-Regular",
                "fontSize": 18,
                "marginTop": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          Follow artists to see new works that have been added to Artsy
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
